library/CHART-TEMPLATE/templates/deployment.yaml

```diff
@@ -49,6 +49,11 @@ spec:
               port: http
           resources:
             {{- toYaml .Values.resources | nindent 12 }}
+          env:
+            - name: MY_APP_NON_OPTIONAL_ENV_VAR
```

Comment:
why no value for the env? 


in nri-prometheus for example the approach is different:

https://github.com/newrelic/helm-charts/blob/afd566c86e2d336c08d52edc7ab5a60fae6859b7/charts/nri-prometheus/values.yaml#L73



The user is required to add the url, it is not enough just to enable a flag

Reply:
other chart deal with it in a different way?

Reply:
answered above